src/rust/en.omegascans/src/parser.rs

```diff
@@ -0,0 +1,406 @@
+							genres.push(',');
+						}
+						_ => continue,
+					}
+				}
+			}
+			FilterType::Select => match filter.name.as_str() {
+				"Status" => match filter.value.as_int().unwrap_or(-1) {
+					0 => status.push_str("All"),
+					1 => status.push_str("Ongoing"),
+					2 => status.push_str("Hiatus"),
+					3 => status.push_str("Dropped"),
+					4 => status.push_str("Completed"),
+					5 => status.push_str("Cancelled"),
+					_ => continue,
+				},
+				_ => continue,
+			}
+			_ => continue,
+		}
+	}
+
+	if !genres.is_empty() {
+		genres.pop();
+	}
+
+	let url = format!("https://api.omegascans.org/query?query_string={}&series_status={}&order=desc&orderBy=total_views&series_type=Comic&page=1&perPage=1000&tags_ids=[{}]", search_query, status, genres);
+
+	let json = Request::new(url, HttpMethod::Get)
+		.json()
+		.expect("Failed to load JSON")
+		.as_object()
+		.expect("Failed to get JSON as object");
+
+	let data = json
+		.get("data")
+		.as_array()
+		.expect("Failed to get data as object");
+
+	let mut mangas: Vec<Manga> = Vec::new();
+
+	for manga in data {
+		let manga = manga.as_object().expect("Failed to get manga as object");
+
+		let title = manga
+			.get("title")
+			.as_string()
+			.expect("Failed to get manga name as str")
+			.read();
+
+		let cover = manga
+			.get("thumbnail")
+			.as_string()
+			.expect("Failed to get manga cover as str")
+			.read();
+
+		let slug = manga
+			.get("series_slug")
+			.as_string()
+			.expect("Failed to get manga slug as str")
+			.read();
+
+		let status = manga
+			.get("status")
+			.as_string()
+			.expect("Failed to get manga status as array")
+			.read();
+
+		let mut manga_status = MangaStatus::Unknown;
+
+		match status.as_str() {
+			"New" => {}
+			"Ongoing" => manga_status = MangaStatus::Ongoing,
+			"Completed" => manga_status = MangaStatus::Completed,
+			"Dropped" => manga_status = MangaStatus::Cancelled,
+			"Hiatus" => manga_status = MangaStatus::Hiatus,
+			_ => manga_status = MangaStatus::Unknown,
+		};
+
+
+		let url = format!("{}/series/{}", base_url, slug);
+
+		mangas.push(Manga {
+			id: slug,
+			cover: cover,
+			title: title,
+			author: String::new(),
+			artist: String::new(),
+			description: String::new(),
+			url: url,
+			categories: Vec::new(),
+			status: manga_status,
+			nsfw: MangaContentRating::Nsfw,
+			viewer: MangaViewer::Scroll,
+		});
+	}
+
+	Ok(MangaPageResult {
+		manga: mangas,
+		has_more: false,
+	})
+}
+
+pub fn parse_manga_listing(base_url: String, url: String, _listing: Listing, _page: i32) -> Result<MangaPageResult> {
+	
+	let json = Request::new(url, HttpMethod::Get)
+		.json()
+		.expect("Failed to load JSON")
+		.as_object()
+		.expect("Failed to get JSON as object");
+
+	let data = json
+		.get("data")
+		.as_array()
+		.expect("Failed to get data as object");
+
+	let mut mangas: Vec<Manga> = Vec::new();
+
+	for manga in data {
+		let manga = manga.as_object().expect("Failed to get manga as object");
```

Comment:
looks like you're parsing it the same way as in `parse_manga_list`, so you could unify it and avoid less code duplication